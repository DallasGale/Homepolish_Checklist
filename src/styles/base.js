import * as colors from './colors';
import * as fonts from './fonts';

const base = `
  html,
  body,
  #root {
    height: 100%;
  }

  body {
    font-family: ${fonts.LATO};
    background: ${colors.WHITE};
  }
`;

export default base;
