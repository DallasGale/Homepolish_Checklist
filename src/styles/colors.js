export const WHITE = '#fff';
export const BLACK = '#4e4e4e';
export const THEME_LIGHT_GREY = '#fafafa';
export const THEME_MEDIUM_GREY = '#e9e9e9';
export const THEME_NOT_STARTED = '#f2c7bc';
export const THEME_COMPLETED = '#aaaaaa';
