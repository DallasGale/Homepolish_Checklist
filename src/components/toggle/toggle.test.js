import React from 'react';
import Toggle, { StyledButtonBorder, StyledButtonInner } from './toggle.component';
import * as colors from '../../styles/colors';
import { shallow, render } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components'

const white = colors.WHITE;
const completeTheme = colors.THEME_COMPLETED;
const notStartedTheme = colors.THEME_NOT_STARTED;

const border = shallow(<StyledButtonBorder />)
const inner = shallow(<StyledButtonInner />);

describe('Toggle', () => {
    it('will render without exploding!', () => {
        const tree = renderer.create(<Toggle />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    describe('when task status is not set to "complete"', () => {
        it('will have a default "border-color"', () => {
            expect(border).toHaveStyleRule('border-color', notStartedTheme)
        })
        it('will have "white" inner "background-color"', () => {
            expect(inner).toHaveStyleRule('background-color', white)
        })
    })

    describe('when task status is set to "complete"', () => {
        
        describe("<StyledButtonInner />", () => {
            it('will change "border-color"', () => {
                border.setProps({ 
                    taskStatus: 'complete'
                })
                expect(border).toHaveStyleRule('border-color', completeTheme)
            })
        });

        describe("<StyledButtonInner />", () => {
            it('will change "background-color"', () => {
                inner.setProps({ 
                    taskStatus: 'complete' 
                })
                expect(inner).toHaveStyleRule('background-color', completeTheme)
            })
        })
    })

})