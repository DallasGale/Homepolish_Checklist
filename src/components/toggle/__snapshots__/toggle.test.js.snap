// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle will render without exploding! 1`] = `
.c1 {
  background: white;
  width: 20px;
  height: 20px;
  border-radius: 20px;
  border: 1px solid;
  border-color: #f2c7bc;
  position: relative;
  margin: 0 auto;
}

.c2 {
  background-color: #fff;
  width: 18px;
  height: 18px;
  border-radius: 15px;
  margin: 0 auto;
  position: relative;
  top: 1px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c2:hover {
  background-color: #f2c7bc;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  border: 0;
  background: none;
  cursor: pointer;
  text-align: center;
  width: 20px;
  height: 20px;
  padding: 0;
  position: relative;
}

<button
  className="c0"
  onClick={undefined}
  value={undefined}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
  </div>
</button>
`;
